// @azure/arm-workloadorchestration/dist/react-native/classic/configTemplates/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { listBySubscription, listByResourceGroup, $delete, removeVersion, createVersion, update, createOrUpdate, get, } from "../../api/configTemplates/operations.js";
function _getConfigTemplates(context) {
    return {
        listBySubscription: (options) => listBySubscription(context, options),
        listByResourceGroup: (resourceGroupName, options) => listByResourceGroup(context, resourceGroupName, options),
        delete: (resourceGroupName, configTemplateName, options) => $delete(context, resourceGroupName, configTemplateName, options),
        removeVersion: (resourceGroupName, configTemplateName, body, options) => removeVersion(context, resourceGroupName, configTemplateName, body, options),
        createVersion: (resourceGroupName, configTemplateName, body, options) => createVersion(context, resourceGroupName, configTemplateName, body, options),
        update: (resourceGroupName, configTemplateName, properties, options) => update(context, resourceGroupName, configTemplateName, properties, options),
        createOrUpdate: (resourceGroupName, configTemplateName, resource, options) => createOrUpdate(context, resourceGroupName, configTemplateName, resource, options),
        get: (resourceGroupName, configTemplateName, options) => get(context, resourceGroupName, configTemplateName, options),
    };
}
export function _getConfigTemplatesOperations(context) {
    return {
        ..._getConfigTemplates(context),
    };
}
