// @azure/arm-workloadorchestration/dist/esm/api/schemaReferences/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { errorResponseDeserializer, schemaReferenceDeserializer, _schemaReferenceListResultDeserializer, } from "../../models/models.js";
import { buildPagedAsyncIterator, } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
export function _listByResourceGroupSend(context, resourceUri, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/{+resourceUri}/providers/Microsoft.Edge/schemaReferences{?api%2Dversion}", {
        resourceUri: resourceUri,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _listByResourceGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _schemaReferenceListResultDeserializer(result.body);
}
/** List by specified resource group */
export function listByResourceGroup(context, resourceUri, options = {
    requestOptions: {},
}) {
    return buildPagedAsyncIterator(context, () => _listByResourceGroupSend(context, resourceUri, options), _listByResourceGroupDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
export function _getSend(context, resourceUri, schemaReferenceName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/{+resourceUri}/providers/Microsoft.Edge/schemaReferences/{schemaReferenceName}{?api%2Dversion}", {
        resourceUri: resourceUri,
        schemaReferenceName: schemaReferenceName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return schemaReferenceDeserializer(result.body);
}
/** Get a Schema Reference Resource */
export async function get(context, resourceUri, schemaReferenceName, options = { requestOptions: {} }) {
    const result = await _getSend(context, resourceUri, schemaReferenceName, options);
    return _getDeserialize(result);
}
