// @azure/arm-workloadorchestration/dist/browser/api/instanceHistories/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { errorResponseDeserializer, instanceHistoryDeserializer, _instanceHistoryListResultDeserializer, } from "../../models/models.js";
import { buildPagedAsyncIterator, } from "../../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
export function _listByInstanceSend(context, resourceGroupName, targetName, solutionName, instanceName, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/solutions/{solutionName}/instances/{instanceName}/histories{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        solutionName: solutionName,
        instanceName: instanceName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _listByInstanceDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _instanceHistoryListResultDeserializer(result.body);
}
/** List InstanceHistory Resources */
export function listByInstance(context, resourceGroupName, targetName, solutionName, instanceName, options = {
    requestOptions: {},
}) {
    return buildPagedAsyncIterator(context, () => _listByInstanceSend(context, resourceGroupName, targetName, solutionName, instanceName, options), _listByInstanceDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
export function _getSend(context, resourceGroupName, targetName, solutionName, instanceName, instanceHistoryName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/solutions/{solutionName}/instances/{instanceName}/histories/{instanceHistoryName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        solutionName: solutionName,
        instanceName: instanceName,
        instanceHistoryName: instanceHistoryName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return instanceHistoryDeserializer(result.body);
}
/** Get InstanceHistory Resource */
export async function get(context, resourceGroupName, targetName, solutionName, instanceName, instanceHistoryName, options = { requestOptions: {} }) {
    const result = await _getSend(context, resourceGroupName, targetName, solutionName, instanceName, instanceHistoryName, options);
    return _getDeserialize(result);
}
