// @azure/arm-workloadorchestration/dist/browser/classic/schemas/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { listBySubscription, listByResourceGroup, removeVersion, createVersion, $delete, update, createOrUpdate, get, } from "../../api/schemas/operations.js";
function _getSchemas(context) {
    return {
        listBySubscription: (options) => listBySubscription(context, options),
        listByResourceGroup: (resourceGroupName, options) => listByResourceGroup(context, resourceGroupName, options),
        removeVersion: (resourceGroupName, schemaName, body, options) => removeVersion(context, resourceGroupName, schemaName, body, options),
        createVersion: (resourceGroupName, schemaName, body, options) => createVersion(context, resourceGroupName, schemaName, body, options),
        delete: (resourceGroupName, schemaName, options) => $delete(context, resourceGroupName, schemaName, options),
        update: (resourceGroupName, schemaName, properties, options) => update(context, resourceGroupName, schemaName, properties, options),
        createOrUpdate: (resourceGroupName, schemaName, resource, options) => createOrUpdate(context, resourceGroupName, schemaName, resource, options),
        get: (resourceGroupName, schemaName, options) => get(context, resourceGroupName, schemaName, options),
    };
}
export function _getSchemasOperations(context) {
    return {
        ..._getSchemas(context),
    };
}
