// @azure/arm-workloadorchestration/dist/commonjs/classic/schemas/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._getSchemasOperations = _getSchemasOperations;
const operations_js_1 = require("../../api/schemas/operations.js");
function _getSchemas(context) {
    return {
        listBySubscription: (options) => (0, operations_js_1.listBySubscription)(context, options),
        listByResourceGroup: (resourceGroupName, options) => (0, operations_js_1.listByResourceGroup)(context, resourceGroupName, options),
        removeVersion: (resourceGroupName, schemaName, body, options) => (0, operations_js_1.removeVersion)(context, resourceGroupName, schemaName, body, options),
        createVersion: (resourceGroupName, schemaName, body, options) => (0, operations_js_1.createVersion)(context, resourceGroupName, schemaName, body, options),
        delete: (resourceGroupName, schemaName, options) => (0, operations_js_1.$delete)(context, resourceGroupName, schemaName, options),
        update: (resourceGroupName, schemaName, properties, options) => (0, operations_js_1.update)(context, resourceGroupName, schemaName, properties, options),
        createOrUpdate: (resourceGroupName, schemaName, resource, options) => (0, operations_js_1.createOrUpdate)(context, resourceGroupName, schemaName, resource, options),
        get: (resourceGroupName, schemaName, options) => (0, operations_js_1.get)(context, resourceGroupName, schemaName, options),
    };
}
function _getSchemasOperations(context) {
    return {
        ..._getSchemas(context),
    };
}
