// @azure/arm-workloadorchestration/dist/commonjs/workloadOrchestrationManagementClient.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadOrchestrationManagementClient = void 0;
const index_js_1 = require("./api/index.js");
const index_js_2 = require("./classic/configTemplateVersions/index.js");
const index_js_3 = require("./classic/configTemplates/index.js");
const index_js_4 = require("./classic/contexts/index.js");
const index_js_5 = require("./classic/diagnostics/index.js");
const index_js_6 = require("./classic/dynamicSchemaVersions/index.js");
const index_js_7 = require("./classic/dynamicSchemas/index.js");
const index_js_8 = require("./classic/executions/index.js");
const index_js_9 = require("./classic/instanceHistories/index.js");
const index_js_10 = require("./classic/instances/index.js");
const index_js_11 = require("./classic/jobs/index.js");
const index_js_12 = require("./classic/schemaReferences/index.js");
const index_js_13 = require("./classic/schemaVersions/index.js");
const index_js_14 = require("./classic/schemas/index.js");
const index_js_15 = require("./classic/siteReferences/index.js");
const index_js_16 = require("./classic/solutionTemplateVersions/index.js");
const index_js_17 = require("./classic/solutionTemplates/index.js");
const index_js_18 = require("./classic/solutionVersions/index.js");
const index_js_19 = require("./classic/solutions/index.js");
const index_js_20 = require("./classic/targets/index.js");
const index_js_21 = require("./classic/workflowVersions/index.js");
const index_js_22 = require("./classic/workflows/index.js");
class WorkloadOrchestrationManagementClient {
    _client;
    /** The pipeline used by this client to make requests */
    pipeline;
    /** Microsoft.Edge Resource Provider management API. */
    constructor(credential, subscriptionId, options = {}) {
        const prefixFromOptions = options?.userAgentOptions?.userAgentPrefix;
        const userAgentPrefix = prefixFromOptions
            ? `${prefixFromOptions} azsdk-js-client`
            : `azsdk-js-client`;
        this._client = (0, index_js_1.createWorkloadOrchestrationManagement)(credential, subscriptionId, {
            ...options,
            userAgentOptions: { userAgentPrefix },
        });
        this.pipeline = this._client.pipeline;
        this.siteReferences = (0, index_js_15._getSiteReferencesOperations)(this._client);
        this.contexts = (0, index_js_4._getContextsOperations)(this._client);
        this.diagnostics = (0, index_js_5._getDiagnosticsOperations)(this._client);
        this.executions = (0, index_js_8._getExecutionsOperations)(this._client);
        this.workflowVersions = (0, index_js_21._getWorkflowVersionsOperations)(this._client);
        this.workflows = (0, index_js_22._getWorkflowsOperations)(this._client);
        this.configTemplateVersions = (0, index_js_2._getConfigTemplateVersionsOperations)(this._client);
        this.configTemplates = (0, index_js_3._getConfigTemplatesOperations)(this._client);
        this.instanceHistories = (0, index_js_9._getInstanceHistoriesOperations)(this._client);
        this.instances = (0, index_js_10._getInstancesOperations)(this._client);
        this.solutionTemplates = (0, index_js_17._getSolutionTemplatesOperations)(this._client);
        this.solutionTemplateVersions = (0, index_js_16._getSolutionTemplateVersionsOperations)(this._client);
        this.solutions = (0, index_js_19._getSolutionsOperations)(this._client);
        this.schemaReferences = (0, index_js_12._getSchemaReferencesOperations)(this._client);
        this.dynamicSchemaVersions = (0, index_js_6._getDynamicSchemaVersionsOperations)(this._client);
        this.targets = (0, index_js_20._getTargetsOperations)(this._client);
        this.jobs = (0, index_js_11._getJobsOperations)(this._client);
        this.solutionVersions = (0, index_js_18._getSolutionVersionsOperations)(this._client);
        this.schemaVersions = (0, index_js_13._getSchemaVersionsOperations)(this._client);
        this.schemas = (0, index_js_14._getSchemasOperations)(this._client);
        this.dynamicSchemas = (0, index_js_7._getDynamicSchemasOperations)(this._client);
    }
    /** The operation groups for siteReferences */
    siteReferences;
    /** The operation groups for contexts */
    contexts;
    /** The operation groups for diagnostics */
    diagnostics;
    /** The operation groups for executions */
    executions;
    /** The operation groups for workflowVersions */
    workflowVersions;
    /** The operation groups for workflows */
    workflows;
    /** The operation groups for configTemplateVersions */
    configTemplateVersions;
    /** The operation groups for configTemplates */
    configTemplates;
    /** The operation groups for instanceHistories */
    instanceHistories;
    /** The operation groups for instances */
    instances;
    /** The operation groups for solutionTemplates */
    solutionTemplates;
    /** The operation groups for solutionTemplateVersions */
    solutionTemplateVersions;
    /** The operation groups for solutions */
    solutions;
    /** The operation groups for schemaReferences */
    schemaReferences;
    /** The operation groups for dynamicSchemaVersions */
    dynamicSchemaVersions;
    /** The operation groups for targets */
    targets;
    /** The operation groups for jobs */
    jobs;
    /** The operation groups for solutionVersions */
    solutionVersions;
    /** The operation groups for schemaVersions */
    schemaVersions;
    /** The operation groups for schemas */
    schemas;
    /** The operation groups for dynamicSchemas */
    dynamicSchemas;
}
exports.WorkloadOrchestrationManagementClient = WorkloadOrchestrationManagementClient;
