// @azure/arm-workloadorchestration/dist/commonjs/static-helpers/pollingHelpers.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLongRunningPoller = getLongRunningPoller;
const core_lro_1 = require("@azure/core-lro");
const core_client_1 = require("@azure-rest/core-client");
function getLongRunningPoller(client, processResponseBody, expectedStatuses, options) {
    const { restoreFrom, getInitialResponse } = options;
    if (!restoreFrom && !getInitialResponse) {
        throw new Error("Either restoreFrom or getInitialResponse must be specified");
    }
    let initialResponse = undefined;
    const pollAbortController = new AbortController();
    const poller = {
        sendInitialRequest: async () => {
            if (!getInitialResponse) {
                throw new Error("getInitialResponse is required when initializing a new poller");
            }
            initialResponse = await getInitialResponse();
            return getLroResponse(initialResponse, expectedStatuses);
        },
        sendPollRequest: async (path, pollOptions) => {
            // The poll request would both listen to the user provided abort signal and the poller's own abort signal
            function abortListener() {
                pollAbortController.abort();
            }
            const abortSignal = pollAbortController.signal;
            if (options.abortSignal?.aborted) {
                pollAbortController.abort();
            }
            else if (pollOptions?.abortSignal?.aborted) {
                pollAbortController.abort();
            }
            else if (!abortSignal.aborted) {
                options.abortSignal?.addEventListener("abort", abortListener, {
                    once: true,
                });
                pollOptions?.abortSignal?.addEventListener("abort", abortListener, {
                    once: true,
                });
            }
            let response;
            try {
                response = await client.pathUnchecked(path).get({ abortSignal });
            }
            finally {
                options.abortSignal?.removeEventListener("abort", abortListener);
                pollOptions?.abortSignal?.removeEventListener("abort", abortListener);
            }
            return getLroResponse(response, expectedStatuses);
        },
    };
    return (0, core_lro_1.createHttpPoller)(poller, {
        intervalInMs: options?.updateIntervalInMs,
        resourceLocationConfig: options?.resourceLocationConfig,
        restoreFrom: options?.restoreFrom,
        processResult: (result) => {
            return processResponseBody(result);
        },
    });
}
/**
 * Converts a Rest Client response to a response that the LRO implementation understands
 * @param response - a rest client http response
 * @param deserializeFn - deserialize function to convert Rest response to modular output
 * @returns - An LRO response that the LRO implementation understands
 */
function getLroResponse(response, expectedStatuses) {
    if (!expectedStatuses.includes(response.status)) {
        throw (0, core_client_1.createRestError)(response);
    }
    return {
        flatResponse: response,
        rawResponse: {
            ...response,
            statusCode: Number.parseInt(response.status),
            body: response.body,
        },
    };
}
