// @azure/arm-workloadorchestration/dist/commonjs/api/configTemplateVersions/operations.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._listByConfigTemplateSend = _listByConfigTemplateSend;
exports._listByConfigTemplateDeserialize = _listByConfigTemplateDeserialize;
exports.listByConfigTemplate = listByConfigTemplate;
exports._getSend = _getSend;
exports._getDeserialize = _getDeserialize;
exports.get = get;
const models_js_1 = require("../../models/models.js");
const pagingHelpers_js_1 = require("../../static-helpers/pagingHelpers.js");
const urlTemplate_js_1 = require("../../static-helpers/urlTemplate.js");
const core_client_1 = require("@azure-rest/core-client");
function _listByConfigTemplateSend(context, resourceGroupName, configTemplateName, options = {
    requestOptions: {},
}) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/configTemplates/{configTemplateName}/versions{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        configTemplateName: configTemplateName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
async function _listByConfigTemplateDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1._configTemplateVersionListResultDeserializer)(result.body);
}
/** List Config Template Version Resources */
function listByConfigTemplate(context, resourceGroupName, configTemplateName, options = {
    requestOptions: {},
}) {
    return (0, pagingHelpers_js_1.buildPagedAsyncIterator)(context, () => _listByConfigTemplateSend(context, resourceGroupName, configTemplateName, options), _listByConfigTemplateDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
function _getSend(context, resourceGroupName, configTemplateName, configTemplateVersionName, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/configTemplates/{configTemplateName}/versions/{configTemplateVersionName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        configTemplateName: configTemplateName,
        configTemplateVersionName: configTemplateVersionName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.configTemplateVersionDeserializer)(result.body);
}
/** Get a Config Template Version Resource */
async function get(context, resourceGroupName, configTemplateName, configTemplateVersionName, options = { requestOptions: {} }) {
    const result = await _getSend(context, resourceGroupName, configTemplateName, configTemplateVersionName, options);
    return _getDeserialize(result);
}
