// @azure/arm-workloadorchestration/dist/commonjs/api/targets/operations.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._updateExternalValidationStatusSend = _updateExternalValidationStatusSend;
exports._updateExternalValidationStatusDeserialize = _updateExternalValidationStatusDeserialize;
exports.updateExternalValidationStatus = updateExternalValidationStatus;
exports._publishSolutionVersionSend = _publishSolutionVersionSend;
exports._publishSolutionVersionDeserialize = _publishSolutionVersionDeserialize;
exports.publishSolutionVersion = publishSolutionVersion;
exports._reviewSolutionVersionSend = _reviewSolutionVersionSend;
exports._reviewSolutionVersionDeserialize = _reviewSolutionVersionDeserialize;
exports.reviewSolutionVersion = reviewSolutionVersion;
exports._resolveConfigurationSend = _resolveConfigurationSend;
exports._resolveConfigurationDeserialize = _resolveConfigurationDeserialize;
exports.resolveConfiguration = resolveConfiguration;
exports._removeRevisionSend = _removeRevisionSend;
exports._removeRevisionDeserialize = _removeRevisionDeserialize;
exports.removeRevision = removeRevision;
exports._uninstallSolutionSend = _uninstallSolutionSend;
exports._uninstallSolutionDeserialize = _uninstallSolutionDeserialize;
exports.uninstallSolution = uninstallSolution;
exports._installSolutionSend = _installSolutionSend;
exports._installSolutionDeserialize = _installSolutionDeserialize;
exports.installSolution = installSolution;
exports._listBySubscriptionSend = _listBySubscriptionSend;
exports._listBySubscriptionDeserialize = _listBySubscriptionDeserialize;
exports.listBySubscription = listBySubscription;
exports._listByResourceGroupSend = _listByResourceGroupSend;
exports._listByResourceGroupDeserialize = _listByResourceGroupDeserialize;
exports.listByResourceGroup = listByResourceGroup;
exports._$deleteSend = _$deleteSend;
exports._$deleteDeserialize = _$deleteDeserialize;
exports.$delete = $delete;
exports._updateSend = _updateSend;
exports._updateDeserialize = _updateDeserialize;
exports.update = update;
exports._createOrUpdateSend = _createOrUpdateSend;
exports._createOrUpdateDeserialize = _createOrUpdateDeserialize;
exports.createOrUpdate = createOrUpdate;
exports._getSend = _getSend;
exports._getDeserialize = _getDeserialize;
exports.get = get;
const models_js_1 = require("../../models/models.js");
const pagingHelpers_js_1 = require("../../static-helpers/pagingHelpers.js");
const pollingHelpers_js_1 = require("../../static-helpers/pollingHelpers.js");
const urlTemplate_js_1 = require("../../static-helpers/urlTemplate.js");
const core_client_1 = require("@azure-rest/core-client");
function _updateExternalValidationStatusSend(context, resourceGroupName, targetName, body, options = {
    requestOptions: {},
}) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/updateExternalValidationStatus{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.updateExternalValidationStatusParameterSerializer)(body),
    });
}
async function _updateExternalValidationStatusDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.solutionVersionDeserializer)(result.body);
}
/** Post request to update external validation status */
function updateExternalValidationStatus(context, resourceGroupName, targetName, body, options = {
    requestOptions: {},
}) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _updateExternalValidationStatusDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _updateExternalValidationStatusSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _publishSolutionVersionSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/publishSolutionVersion{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.solutionVersionParameterSerializer)(body),
    });
}
async function _publishSolutionVersionDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.solutionVersionDeserializer)(result.body);
}
/** Post request to publish */
function publishSolutionVersion(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _publishSolutionVersionDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _publishSolutionVersionSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _reviewSolutionVersionSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/reviewSolutionVersion{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.solutionTemplateParameterSerializer)(body),
    });
}
async function _reviewSolutionVersionDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.solutionVersionDeserializer)(result.body);
}
/** Post request to review configuration */
function reviewSolutionVersion(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _reviewSolutionVersionDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _reviewSolutionVersionSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _resolveConfigurationSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/resolveConfiguration{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.solutionTemplateParameterSerializer)(body),
    });
}
async function _resolveConfigurationDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.resolvedConfigurationDeserializer)(result.body);
}
/** Post request to resolve configuration */
function resolveConfiguration(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _resolveConfigurationDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _resolveConfigurationSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _removeRevisionSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/removeRevision{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        body: (0, models_js_1.removeRevisionParameterSerializer)(body),
    });
}
async function _removeRevisionDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return;
}
/** Post request to remove solution version revision */
function removeRevision(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _removeRevisionDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _removeRevisionSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _uninstallSolutionSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/uninstallSolution{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        body: (0, models_js_1.uninstallSolutionParameterSerializer)(body),
    });
}
async function _uninstallSolutionDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return;
}
/** Post request to uninstall */
function uninstallSolution(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _uninstallSolutionDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _uninstallSolutionSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _installSolutionSend(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}/installSolution{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).post({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        body: (0, models_js_1.installSolutionParameterSerializer)(body),
    });
}
async function _installSolutionDeserialize(result) {
    const expectedStatuses = ["202", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return;
}
/** Post request to deploy */
function installSolution(context, resourceGroupName, targetName, body, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _installSolutionDeserialize, ["202", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _installSolutionSend(context, resourceGroupName, targetName, body, options),
        resourceLocationConfig: "location",
    });
}
function _listBySubscriptionSend(context, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/providers/Microsoft.Edge/targets{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
async function _listBySubscriptionDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1._targetListResultDeserializer)(result.body);
}
/** List by subscription */
function listBySubscription(context, options = { requestOptions: {} }) {
    return (0, pagingHelpers_js_1.buildPagedAsyncIterator)(context, () => _listBySubscriptionSend(context, options), _listBySubscriptionDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
function _listByResourceGroupSend(context, resourceGroupName, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
async function _listByResourceGroupDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1._targetListResultDeserializer)(result.body);
}
/** List by specified resource group */
function listByResourceGroup(context, resourceGroupName, options = { requestOptions: {} }) {
    return (0, pagingHelpers_js_1.buildPagedAsyncIterator)(context, () => _listByResourceGroupSend(context, resourceGroupName, options), _listByResourceGroupDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
function _$deleteSend(context, resourceGroupName, targetName, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}{?api%2Dversion,forceDelete}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
        forceDelete: options?.forceDelete,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...(0, core_client_1.operationOptionsToRequestParameters)(options) });
}
async function _$deleteDeserialize(result) {
    const expectedStatuses = ["202", "204", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return;
}
/** Delete a Target Resource */
/**
 *  @fixme delete is a reserved word that cannot be used as an operation name.
 *         Please add @clientName("clientName") or @clientName("<JS-Specific-Name>", "javascript")
 *         to the operation to override the generated name.
 */
function $delete(context, resourceGroupName, targetName, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _$deleteDeserialize, ["202", "204", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _$deleteSend(context, resourceGroupName, targetName, options),
        resourceLocationConfig: "location",
    });
}
function _updateSend(context, resourceGroupName, targetName, properties, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).patch({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.targetUpdateSerializer)(properties),
    });
}
async function _updateDeserialize(result) {
    const expectedStatuses = ["200", "202"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.targetDeserializer)(result.body);
}
/** update a Target Resource */
function update(context, resourceGroupName, targetName, properties, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _updateDeserialize, ["200", "202"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _updateSend(context, resourceGroupName, targetName, properties, options),
        resourceLocationConfig: "location",
    });
}
function _createOrUpdateSend(context, resourceGroupName, targetName, resource, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: (0, models_js_1.targetSerializer)(resource),
    });
}
async function _createOrUpdateDeserialize(result) {
    const expectedStatuses = ["200", "201", "202"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.targetDeserializer)(result.body);
}
/** Create or update a Target Resource */
function createOrUpdate(context, resourceGroupName, targetName, resource, options = { requestOptions: {} }) {
    return (0, pollingHelpers_js_1.getLongRunningPoller)(context, _createOrUpdateDeserialize, ["200", "201", "202"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _createOrUpdateSend(context, resourceGroupName, targetName, resource, options),
        resourceLocationConfig: "azure-async-operation",
    });
}
function _getSend(context, resourceGroupName, targetName, options = { requestOptions: {} }) {
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/targets/{targetName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        targetName: targetName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...(0, core_client_1.operationOptionsToRequestParameters)(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = (0, core_client_1.createRestError)(result);
        error.details = (0, models_js_1.errorResponseDeserializer)(result.body);
        throw error;
    }
    return (0, models_js_1.targetDeserializer)(result.body);
}
/** Get a Target Resource */
async function get(context, resourceGroupName, targetName, options = { requestOptions: {} }) {
    const result = await _getSend(context, resourceGroupName, targetName, options);
    return _getDeserialize(result);
}
