// @azure/arm-workloadorchestration/dist/browser/classic/solutionTemplates/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { listBySubscription, listByResourceGroup, $delete, removeVersion, createVersion, update, createOrUpdate, get, } from "../../api/solutionTemplates/operations.js";
function _getSolutionTemplates(context) {
    return {
        listBySubscription: (options) => listBySubscription(context, options),
        listByResourceGroup: (resourceGroupName, options) => listByResourceGroup(context, resourceGroupName, options),
        delete: (resourceGroupName, solutionTemplateName, options) => $delete(context, resourceGroupName, solutionTemplateName, options),
        removeVersion: (resourceGroupName, solutionTemplateName, body, options) => removeVersion(context, resourceGroupName, solutionTemplateName, body, options),
        createVersion: (resourceGroupName, solutionTemplateName, body, options) => createVersion(context, resourceGroupName, solutionTemplateName, body, options),
        update: (resourceGroupName, solutionTemplateName, properties, options) => update(context, resourceGroupName, solutionTemplateName, properties, options),
        createOrUpdate: (resourceGroupName, solutionTemplateName, resource, options) => createOrUpdate(context, resourceGroupName, solutionTemplateName, resource, options),
        get: (resourceGroupName, solutionTemplateName, options) => get(context, resourceGroupName, solutionTemplateName, options),
    };
}
export function _getSolutionTemplatesOperations(context) {
    return {
        ..._getSolutionTemplates(context),
    };
}
