// @azure/arm-workloadorchestration/dist/commonjs/classic/configTemplates/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._getConfigTemplatesOperations = _getConfigTemplatesOperations;
const operations_js_1 = require("../../api/configTemplates/operations.js");
function _getConfigTemplates(context) {
    return {
        listBySubscription: (options) => (0, operations_js_1.listBySubscription)(context, options),
        listByResourceGroup: (resourceGroupName, options) => (0, operations_js_1.listByResourceGroup)(context, resourceGroupName, options),
        delete: (resourceGroupName, configTemplateName, options) => (0, operations_js_1.$delete)(context, resourceGroupName, configTemplateName, options),
        removeVersion: (resourceGroupName, configTemplateName, body, options) => (0, operations_js_1.removeVersion)(context, resourceGroupName, configTemplateName, body, options),
        createVersion: (resourceGroupName, configTemplateName, body, options) => (0, operations_js_1.createVersion)(context, resourceGroupName, configTemplateName, body, options),
        update: (resourceGroupName, configTemplateName, properties, options) => (0, operations_js_1.update)(context, resourceGroupName, configTemplateName, properties, options),
        createOrUpdate: (resourceGroupName, configTemplateName, resource, options) => (0, operations_js_1.createOrUpdate)(context, resourceGroupName, configTemplateName, resource, options),
        get: (resourceGroupName, configTemplateName, options) => (0, operations_js_1.get)(context, resourceGroupName, configTemplateName, options),
    };
}
function _getConfigTemplatesOperations(context) {
    return {
        ..._getConfigTemplates(context),
    };
}
