// @azure/arm-workloadorchestration/dist/react-native/api/dynamicSchemaVersions/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { errorResponseDeserializer, dynamicSchemaVersionSerializer, dynamicSchemaVersionDeserializer, _dynamicSchemaVersionListResultDeserializer, } from "../../models/models.js";
import { buildPagedAsyncIterator, } from "../../static-helpers/pagingHelpers.js";
import { getLongRunningPoller } from "../../static-helpers/pollingHelpers.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
export function _listByDynamicSchemaSend(context, resourceGroupName, schemaName, dynamicSchemaName, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/schemas/{schemaName}/dynamicSchemas/{dynamicSchemaName}/versions{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        schemaName: schemaName,
        dynamicSchemaName: dynamicSchemaName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _listByDynamicSchemaDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return _dynamicSchemaVersionListResultDeserializer(result.body);
}
/** List by Dynamic Schema */
export function listByDynamicSchema(context, resourceGroupName, schemaName, dynamicSchemaName, options = {
    requestOptions: {},
}) {
    return buildPagedAsyncIterator(context, () => _listByDynamicSchemaSend(context, resourceGroupName, schemaName, dynamicSchemaName, options), _listByDynamicSchemaDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
export function _$deleteSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/schemas/{schemaName}/dynamicSchemas/{dynamicSchemaName}/versions/{dynamicSchemaVersionName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        schemaName: schemaName,
        dynamicSchemaName: dynamicSchemaName,
        dynamicSchemaVersionName: dynamicSchemaVersionName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).delete({ ...operationOptionsToRequestParameters(options) });
}
export async function _$deleteDeserialize(result) {
    const expectedStatuses = ["202", "204", "200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return;
}
/** Delete a Dynamic Schema Version Resource */
/**
 *  @fixme delete is a reserved word that cannot be used as an operation name.
 *         Please add @clientName("clientName") or @clientName("<JS-Specific-Name>", "javascript")
 *         to the operation to override the generated name.
 */
export function $delete(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options = { requestOptions: {} }) {
    return getLongRunningPoller(context, _$deleteDeserialize, ["202", "204", "200"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _$deleteSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options),
        resourceLocationConfig: "location",
    });
}
export function _updateSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, properties, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/schemas/{schemaName}/dynamicSchemas/{dynamicSchemaName}/versions/{dynamicSchemaVersionName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        schemaName: schemaName,
        dynamicSchemaName: dynamicSchemaName,
        dynamicSchemaVersionName: dynamicSchemaVersionName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).patch({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: dynamicSchemaVersionSerializer(properties),
    });
}
export async function _updateDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return dynamicSchemaVersionDeserializer(result.body);
}
/** update a Dynamic Schema Version Resource */
export async function update(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, properties, options = { requestOptions: {} }) {
    const result = await _updateSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, properties, options);
    return _updateDeserialize(result);
}
export function _createOrUpdateSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, resource, options = {
    requestOptions: {},
}) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/schemas/{schemaName}/dynamicSchemas/{dynamicSchemaName}/versions/{dynamicSchemaVersionName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        schemaName: schemaName,
        dynamicSchemaName: dynamicSchemaName,
        dynamicSchemaVersionName: dynamicSchemaVersionName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).put({
        ...operationOptionsToRequestParameters(options),
        contentType: "application/json",
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
        body: dynamicSchemaVersionSerializer(resource),
    });
}
export async function _createOrUpdateDeserialize(result) {
    const expectedStatuses = ["200", "201", "202"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return dynamicSchemaVersionDeserializer(result.body);
}
/** Create or update a Dynamic Schema Version Resource */
export function createOrUpdate(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, resource, options = {
    requestOptions: {},
}) {
    return getLongRunningPoller(context, _createOrUpdateDeserialize, ["200", "201", "202"], {
        updateIntervalInMs: options?.updateIntervalInMs,
        abortSignal: options?.abortSignal,
        getInitialResponse: () => _createOrUpdateSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, resource, options),
        resourceLocationConfig: "azure-async-operation",
    });
}
export function _getSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options = { requestOptions: {} }) {
    const path = expandUrlTemplate("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Edge/schemas/{schemaName}/dynamicSchemas/{dynamicSchemaName}/versions/{dynamicSchemaVersionName}{?api%2Dversion}", {
        subscriptionId: context.subscriptionId,
        resourceGroupName: resourceGroupName,
        schemaName: schemaName,
        dynamicSchemaName: dynamicSchemaName,
        dynamicSchemaVersionName: dynamicSchemaVersionName,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: options?.requestOptions?.skipUrlEncoding,
    });
    return context.path(path).get({
        ...operationOptionsToRequestParameters(options),
        headers: {
            accept: "application/json",
            ...options.requestOptions?.headers,
        },
    });
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        const error = createRestError(result);
        error.details = errorResponseDeserializer(result.body);
        throw error;
    }
    return dynamicSchemaVersionDeserializer(result.body);
}
/** Get a Dynamic Schema Version Resource */
export async function get(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options = { requestOptions: {} }) {
    const result = await _getSend(context, resourceGroupName, schemaName, dynamicSchemaName, dynamicSchemaVersionName, options);
    return _getDeserialize(result);
}
