// @azure/arm-workloadorchestration/dist/commonjs/classic/solutionTemplates/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._getSolutionTemplatesOperations = _getSolutionTemplatesOperations;
const operations_js_1 = require("../../api/solutionTemplates/operations.js");
function _getSolutionTemplates(context) {
    return {
        listBySubscription: (options) => (0, operations_js_1.listBySubscription)(context, options),
        listByResourceGroup: (resourceGroupName, options) => (0, operations_js_1.listByResourceGroup)(context, resourceGroupName, options),
        delete: (resourceGroupName, solutionTemplateName, options) => (0, operations_js_1.$delete)(context, resourceGroupName, solutionTemplateName, options),
        removeVersion: (resourceGroupName, solutionTemplateName, body, options) => (0, operations_js_1.removeVersion)(context, resourceGroupName, solutionTemplateName, body, options),
        createVersion: (resourceGroupName, solutionTemplateName, body, options) => (0, operations_js_1.createVersion)(context, resourceGroupName, solutionTemplateName, body, options),
        update: (resourceGroupName, solutionTemplateName, properties, options) => (0, operations_js_1.update)(context, resourceGroupName, solutionTemplateName, properties, options),
        createOrUpdate: (resourceGroupName, solutionTemplateName, resource, options) => (0, operations_js_1.createOrUpdate)(context, resourceGroupName, solutionTemplateName, resource, options),
        get: (resourceGroupName, solutionTemplateName, options) => (0, operations_js_1.get)(context, resourceGroupName, solutionTemplateName, options),
    };
}
function _getSolutionTemplatesOperations(context) {
    return {
        ..._getSolutionTemplates(context),
    };
}
